cmd/lambda/leases/main.go

```diff
@@ -34,74 +61,117 @@ type messageBody struct {
 	Body    string `json:"Body"`
 }
 
+func init() {
+	log.Println("Cold start; creating router for /leases")
+
+	leaseAddedTopicARN = aws.String(config.GetEnvVar("PROVISION_TOPIC", "DCEDefaultProvisionTopic"))
+	accountDeletedTopicArn = config.GetEnvVar("DECOMMISSION_TOPIC", "DefaultDecomissionTopic")
+	resetQueueURL = config.GetEnvVar("RESET_SQS_URL", "DefaultResetSQSURL")
+	principalBudgetAmount = aws.Float64(config.GetEnvFloatVar("PRINCIPAL_BUDGET_AMOUNT", 1000.00))
+	principalBudgetPeriod = aws.String(config.GetEnvVar("PRINCIPAL_BUDGET_PERIOD", Weekly))
+	maxLeaseBudgetAmount = aws.Float64(config.GetEnvFloatVar("MAX_LEASE_BUDGET_AMOUNT", 1000.00))
+	maxLeasePeriod = aws.Int64(int64(config.GetEnvIntVar("MAX_LEASE_PERIOD", 704800)))
+	defaultLeaseLengthInDays = aws.Int(config.GetEnvIntVar("DEFAULT_LEASE_LENGTH_IN_DAYS", 7))
+
+	leasesRoutes := api.Routes{
```

Comment:
Correct me if I am wrong, but with the new routing logic here for the leases API one can no longer query `/leases?principalId=1&status=Active` - right? These query parameters now looks exclusive.

Reply:
NVM - The way the router works is that if it doesn’t find an exact match on the query params it falls through to the one without